src/components/SearchPage/SearchPage.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Suspense, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { useSwipeable } from "react-swipeable";
+import useVideoSearch from "../../hooks/useVideoSearch";
+import { Video } from "../../types/types";
+import Pagination from "../Pagination";
+import SearchBox from "../SearchBox";
+import "./SearchPage.css";
+
+const VideoList = React.lazy(() => import("../VideoList"));
+const VideoDetail = React.lazy(() => import("../VideoDetail"));
+
+function SearchPage() {
+  const [term, setTerm] = useState<string>("");
+  const [videos, setVideos] = useState<Video[]>([]);
+  const [selectedVideo, setSelectedVideo] = useState<Video | null>(null);
+  const [currentPage, setCurrentPage] = useState<number>(0);
+  const [pageCount, setPageCount] = useState<number>(1);
+  const [offset, setOffset] = useState<number>(4);
+  const [videosPerPage] = useState<number>(4);
+
+  const { data, handleSubmit } = useVideoSearch(term);
+
+  const { isLoading, error } = useQuery<Video[], Error>(
+    "videos",
+    () => handleSubmit(),
+    { enabled: false }
+  );
+
+  const handlePageClick = (e: { selected: number }) => {
+    const selectedPage = e.selected;
+    setCurrentPage(selectedPage);
+    setOffset((selectedPage + 1) * videosPerPage);
+  };
+
+  const handlers = useSwipeable({
+    onSwipedLeft: () => {
+      if (currentPage === pageCount - 1) return;
```

Comment:
This could be also use as a slightly shorter version
if (currentPage !== pageCount - 1) { handlePageClick }